### PR TITLE
feat: replace logic in OsmPbfLoader from osmium to QuackOSM

### PR DESCRIPTION
**Pull request by RaczeQ**

### CODE REVIEW

File: srai/loaders/osm_loaders/filters/_typing.py
Comment (simonusher) [inline srai/loaders/osm_loaders/filters/_typing.py:40]: Should we maybe make this private to simplify the API since the function above handles this?

File: srai/loaders/osm_loaders/osm_pbf_loader.py
Comment (simonusher) [inline srai/loaders/osm_loaders/osm_pbf_loader.py:122]: This whole block seems to be duplicated in here and `load_to_geoparquet`. Maybe extract it into a separate function?

**Review by simonusher — CHANGES_REQUESTED** [review]

File: pyproject.toml
Comment (RaczeQ) [reply pyproject.toml:66]: We are waiting for a release from the maintainers of geoarrow-python https://github.com/geoarrow/geoarrow-python/issues/42
For now I'll lock it to the 0a95d5ff3180c20e89f4572ac38d255f58e13396 commit.

File: srai/loaders/osm_loaders/osm_pbf_loader.py
Comment (RaczeQ) [reply srai/loaders/osm_loaders/osm_pbf_loader.py:122]: You're right, I've simplified the code.

File: srai/loaders/osm_loaders/filters/_typing.py
Comment (RaczeQ) [reply srai/loaders/osm_loaders/filters/_typing.py]: That's a good idea. I've even expanded the functionality of the function.

File: srai/loaders/osm_loaders/pbf_file_downloader.py
Comment (RaczeQ) [reply srai/loaders/osm_loaders/pbf_file_downloader.py:36]: Sure, its functionality was unreliable and janky anyway.

File: srai/loaders/osm_loaders/pbf_file_handler.py
Comment (RaczeQ) [inline srai/loaders/osm_loaders/pbf_file_handler.py:38]: TODO: error when nodes are outside given geometry, but linestring intersects it

Comment (RaczeQ) [reply srai/loaders/osm_loaders/pbf_file_handler.py:38]: Optional: https://gdal.org/drivers/vector/osm.html#spatial-filtering

Comment (RaczeQ) [reply srai/loaders/osm_loaders/pbf_file_handler.py:38]: Create tests based on GDAL results

**Review by piotrgramacki — DISMISSED** [review]
LGTM! Nicely done

**Pull request merged**
